### PR TITLE
Adds a setting to show full values of item counts (instead of >9k)

### PR DESCRIPTION
**Pull request by PKPenguin321**
~~Entirely untested and~~ (mostly) made with web editor

### CODE REVIEW

File: src/components/Content/ContentSettings.vue [hunk 1]
```diff
@@ -80,6 +80,19 @@
...
+            <div class="custom-control custom-switch">
+              <input
+                v-model="showFullValues"
+                type="checkbox"
+                class="custom-control-input"
+                id="showFullValues"
+              />
+              <label
+                class="custom-control-label"
+                for="showVirtualLevels"
```
Comment (TBartl) [inline src/components/Content/ContentSettings.vue:93]:
```suggestion
                for="showFullValues"

```

File: src/components/Content/ContentSettings.vue [hunk 2]
```diff
@@ -80,6 +80,19 @@
...
+              >Show Full Values for Item Counts (can look ugly with huge numbers)</label>
```
Comment (TBartl) [inline src/components/Content/ContentSettings.vue:94]:
```suggestion
              >Display Full Item Counts in Jobs (may look ugly with huge numbers)</label>
```
This doesn't change how item counts are displayed elsewhere (like the money in the top left) so specifying that it's for jobs may help with some confusion.

File: src/state/settings.js [hunk 3]
```diff
@@ -14,6 +14,9 @@ const settings = {
 		showVirtualLevels(state) {
 			return state.showVirtualLevels;
 		},
+		showFullValues(state) {
+			return state.showFullValues;
```
Comment (TBartl) [inline src/state/settings.js:18]: You'll also need to define this variable up in the state above.

File: src/main.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ Vue.filter('stat', function (value) {
 })
 Vue.filter('aggressive', function (value) {
 	if (value == undefined) return 0;
-	if (value >= 1000) {
+	if (value >= 1000 && !rootGetters["settings/showFullValues"]) {
 		value = Math.min(Math.floor(value / 1000), 9);
 		return `>${value}k`;
 	}
```
Comment (TBartl) [inline src/main.js:31]: Can't comment down there, but on line 32 you'll want to return `value.toLocaleString()` so you get the commas in there (this piece of code hasn't needed commas yet).

File: src/main.js [hunk 5]
```diff
@@ -25,7 +25,7 @@ Vue.filter('stat', function (value) {
 })
 Vue.filter('aggressive', function (value) {
 	if (value == undefined) return 0;
-	if (value >= 1000) {
+	if (value >= 1000 && !rootGetters["settings/showFullValues"]) {
```
Comment (TBartl) [inline src/main.js:28]: `rootGetters` won't be defined here. I think you can do `store.rootGetters` though (don't quote me on this)

**Comment by PKPenguin321** [comment]
i'm peabrain and deleted the branch by accident

**Comment by PKPenguin321** [comment]
Tested and this is not functional
Setting does not save when changed and item count displays seem to break entirely once you reach 1k of any item
🤷

File: src/components/Content/ContentSettings.vue [hunk 6]
```diff
@@ -80,6 +80,20 @@
                 for="showVirtualLevels"
               >Show Virtual Levels Beyond {{maxLevel}}</label>
             </div>
+            
+            <div class="custom-control custom-switch">
+              <input
+                v-model="showFullValues"
```
Comment (TBartl) [inline src/components/Content/ContentSettings.vue:86]: You need to set up a computed property to bind your `showFullValues` to down below. It should look something like
```
    showFullValues: {
      get() {
        return this.$store.getters["settings/showFullValues"];
      },
      set(value) {
        this.$store.commit("settings/setShowFullValues", value);
      }
    },
```
Look under the other things under `computed` to get an idea here

File: src/main.js [hunk 7]
```diff
@@ -25,11 +25,11 @@ Vue.filter('stat', function (value) {
 })
 Vue.filter('aggressive', function (value) {
 	if (value == undefined) return 0;
-	if (value >= 1000) {
+	if (value >= 1000 && !store.rootGetters["settings/showFullValues"]) {
```
Comment (TBartl) [inline src/main.js:28]: Remember when I said 
image

Well that was a lie. This should work though (actually tested)

```suggestion
	if (value >= 1000 && !store.getters["settings/showFullValues"]) {
```

File: src/state/settings.js [hunk 8]
```diff
@@ -37,6 +41,9 @@ const settings = {
 		setShowVirtualLevels(state, val) {
 			state.showVirtualLevels = val;
 		},
+		showFullValues(state, val) {
```
Comment (TBartl) [inline src/state/settings.js:44]: This is going to have issues being named the same as the getter.

```suggestion
		setShowFullValues(state, val) {

```

**Comment by PKPenguin321** [comment]
Seems to work great now

**Comment by TBartl** [comment]
Should be in 1.3 soon (tm)

**Pull request merged**